### PR TITLE
conditional rendering in carouseWithImage if description is undefined

### PR DESCRIPTION
**Pull request by asya982**
to prevent errors when description is undefined added additional check for item.description

### CODE REVIEW

File: src/components/carousel-with-image/CarouselWithImage.tsx [hunk 1]
```diff
@@ -31,7 +31,9 @@ const CarouselWithImage: FC<CarouselWithImageProps> = ({ items }) => {
       />
       <Box sx={styles.text}>
         <Typography sx={styles.title}>{t(item.title)}</Typography>
-        <Typography sx={styles.description}>{t(item.description)}</Typography>
+        <Typography sx={styles.description}>
+          {t(item.description ?? '')}
```
Comment (maarekua) [inline src/components/carousel-with-image/CarouselWithImage.tsx:35]: What would be translation of empty string?

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/components/carousel-with-image/CarouselWithImage.tsx (hunk 1 above)
Comment (asya982) [reply src/components/carousel-with-image/CarouselWithImage.tsx]: solved

File: src/containers/course-section/CourseSectionContainer.tsx [hunk 2]
```diff
@@ -45,26 +45,19 @@ import {
   CourseResources
 } from '~/types'
 import { styles } from '~/containers/course-section/CourseSectionContainer.styles'
+import CourseSectionsList from '../course-sections-list/CourseSectionsList'
```
Comment (Tolik170) [inline src/containers/course-section/CourseSectionContainer.tsx:48]: use import alias

**Review by GeorgeShvab — APPROVED** [review]

File: src/containers/course-section/CourseSectionContainer.tsx [hunk 3]
```diff
@@ -45,26 +45,19 @@ import {
...
 
-interface SectionProps {
+interface SectionProps
+  extends Pick<
+    Parameters<typeof CourseSectionsList>[0],
```
Comment (Tolik170) [inline src/containers/course-section/CourseSectionContainer.tsx:52]: can you explain how ```
extends Pick<
    Parameters<typeof CourseSectionsList>[0] 
```should work and why you can`t just extend from another interface or define these properties here like before

File: src/types/common/types/common.types.ts [hunk 4]
```diff
@@ -29,3 +29,29 @@ export type Country = {
...
+export type FormNonInputValueChange<Value, Fields> = (
+  key: keyof Fields,
+  value: Value
+) => void
+
+export type UseFormEventHandler<Fields, Event> = (
+  key: keyof Fields
+) => (event: Event) => void
+
+type FormValidationHandler<TData, TValue> = (
```
Comment (Tolik170) [inline src/types/common/types/common.types.ts:48]: If you didn`t use **T** character before generic name in other types, remove it here too

Comment (asya982) [reply src/types/common/types/common.types.ts]: solved

File: src/containers/course-section/CourseSectionContainer.tsx (hunk 3 above)
Comment (asya982) [reply src/containers/course-section/CourseSectionContainer.tsx]: i moved this repeating types to common interface `CourseSectionHandlers` and extend both components from it

File: src/containers/course-section/CourseSectionContainer.tsx (hunk 2 above)
Comment (asya982) [reply src/containers/course-section/CourseSectionContainer.tsx]: deleted this line, because it's unused now

**Review by maarekua — APPROVED** [review]

**Review by D1nviz — APPROVED** [review]

**Review by mhevyk — APPROVED** [review]

**Review by Tolik170 — APPROVED** [review]

**Pull request merged**